### PR TITLE
add OS_TYPE, mkdir once, static launching files, make install macOS

### PR DESCRIPTION
**Pull request by a2line**
Add `OS_TYPE="Win"` in configure to copy the right batch scripts on Windows (thanks Mehdi!).
Make `mkdir bases` only once.
Static files for Linux/Windows/macOS with right exec permissions.
Killall dans `geneweb.command` (Henri83).
Add make install for macOS with icons for .command files and the folder of geneweb (Henri83).
Add checks for piqi and yojson when building api.

### CODE REVIEW

**Comment by a2line** [comment]
Should I merge both commits?

**Comment by ipfix** [comment]
No you should not merge the two commits.

Actually, I wonder how the commit #3515c62 is useful except for the missing execution permissions ? More over, why would we want to print the version of GeneWeb only in the windows version.

**Comment by a2line** [comment]
Version number is not printed, it's “on their [console] windows”
gw-version

+ we talked a few times of the /bases folder that shouldn't be created when launching each times the programs but once at installation.

**Comment by ipfix** [comment]
Thx for the screenshot ! So... I wonder if this is necessary then :)

As for the `mkdir`, actually I'm not a big fan. Because if I remember correctly, in debian, they use another directory to put the bases. So I wonder if geneweb should be distributed with and empty directory (just like the `old`) or not ?

What are pros ?

File: Makefile [hunk 1]
```diff
@@ -126,4 +120,4 @@ distrib:
 	fi
 	cp setup/lang/*.htm $(DESTDIR)/gw/setup/lang/.
 	cp setup/lang/lexicon.txt $(DESTDIR)/gw/setup/lang/.
-	cp -R hd/* $(DESTDIR)/gw/.
+	cp -R hd/* $(DESTDIR)/gw/.
```
Comment (ipfix) [inline Makefile]: Missing endline !!!

**Comment by a2line** [comment]
Fuck it!

File: Makefile (hunk 1 above)
Comment (a2line) [inline Makefile]: :+1:

**Comment by ipfix** [comment]
Merged

**Pull request closed without merging**